src/coreclr/jit/utils.cpp

```diff
@@ -1285,7 +1285,6 @@ void HelperCallProperties::init()
             case CORINFO_HELP_NEWSFAST_FINALIZE:
             case CORINFO_HELP_NEWSFAST_ALIGN8_FINALIZE:
             case CORINFO_HELP_READYTORUN_NEW:
```

Comment:
Should `CORINFO_HELP_BOX` stay here?

Reply:
@jkotas I debated this with someone offline also. I don't think so. This contract states it only throws OOM, which isn't true previously or now. Regardless of what `CORINFO_HELP_BOX` is set to, all implementations eventually call into `JIT_Box` which can throw non-OOM exceptions. My preference here is to ensure the contract is reality so if it can call `JIT_Box`, exceptions are possible. This purity argument could be pushed back against but given the JIT also rarely calls this helper and instead lines it, I am inclined to enforce the purity here.

Reply:
The (fast) `CORINFO_HELP_BOX` should be only ever called on types for which the slow helper does not throw exceptions (except OOM).

Reply:
Agreed, but that is not what this check this doing. It is describing the nature of the JIT helper, which in this case throws. I agree with your sentiment here, it is identical to my offline conversation. My point though is code is about describing what the JIT helper does not what how the JIT is going to call it.

Reply:
Never mind. I'm not finding my argument as convincing as I once did. I'll move it back and document the expectations.